sympy/physics/vector/frame.py

```diff
@@ -780,15 +735,38 @@ def orient_explicit(self, parent, dcm):
         if not isinstance(dcm, MatrixBase):
             raise TypeError("Amounts must be a SymPy Matrix type object.")
 
-        parent_orient_dcm = dcm
-
-        self._dcm(parent, parent_orient_dcm)
+        self._dcm(parent, dcm)
 
         wvec = self._w_diff_dcm(parent)
         self._ang_vel_dict.update({parent: wvec})
         parent._ang_vel_dict.update({self: -wvec})
         self._var_dict = {}
 
+    def orient_dcm(self, parent, dcm):
+        """Sets the orientation of this reference frame relative to another (parent) reference frame
+        using a direction cosine matrix that describes the rotation from the child to the parent.
+
+        Parameters
+        ==========
+
+        parent : ReferenceFrame
+            Reference frame that this reference frame will be rotated relative
+            to.
+        dcm : Matrix, shape(3, 3)
+            Direction cosine matrix that specifies the relative rotation
+            between the two reference frames.
+
+        Warns
+        ======
+
+        UserWarning
+            If the orientation creates a kinematic loop.
+
+
+        """
+
+        self.orient_explicit(parent = parent, dcm = dcm.transpose())
```

Comment:
Could you move the actual implementation from `orient_explicit` to `orient_dcm` because `orient_dcm` will be the new standard? I.e.:
```python
def orient_explicit(self, parent, dcm):
    """..."""
    self.orient_dcm(parent, dcm.T)

def orient_dcm(self, parent, dcm):
    """..."""
    self._dcm(parent, dcm.T)
    ...
```
I noticed that the tests on `orient_explicit` are currently too minimal, so just make sure that the following still works:
```python
from sympy.physics.mechanics import ReferenceFrame, dynamicsymbols
from sympy.matrices.dense import Matrix

cxx, cyy, czz = dynamicsymbols('c_{xx}, c_{yy}, c_{zz}')
cxy, cxz, cyx = dynamicsymbols('c_{xy}, c_{xz}, c_{yx}')
cyz, czx, czy = dynamicsymbols('c_{yz}, c_{zx}, c_{zy}')
dcxx, dcyy, dczz = dynamicsymbols('c_{xx}, c_{yy}, c_{zz}', 1)
dcxy, dcxz, dcyx = dynamicsymbols('c_{xy}, c_{xz}, c_{yx}', 1)
dcyz, dczx, dczy = dynamicsymbols('c_{yz}, c_{zx}, c_{zy}', 1)
A = ReferenceFrame('A')
B = ReferenceFrame('B')
B_C_A = Matrix([[cxx, cxy, cxz],
                [cyx, cyy, cyz],
                [czx, czy, czz]])
B_w_A = ((cyx*dczx + cyy*dczy + cyz*dczz)*B.x +
         (czx*dcxx + czy*dcxy + czz*dcxz)*B.y +
         (cxx*dcyx + cxy*dcyy + cxz*dcyz)*B.z)
A.orient_explicit(B, B_C_A)
assert B.dcm(A) == B_C_A
assert A.ang_vel_in(B) == B_w_A
assert B.ang_vel_in(A) == -B_w_A
```

P.S. You can make a unittest of the code above.